src/main/java/com/pyranid/KotlinDefaultResultSetMapper.kt

```diff
@@ -110,14 +120,95 @@ open class KotlinDefaultResultSetMapper(private val javaDefaultResultSetMapper:
+            if (BigInteger::class.isSuperclassOf(propertyType)) return bigDecimal.toBigInteger()
+        }
+
+        if (resultSetValue is BigInteger) {
+            val bigInteger = resultSetValue
+            if (BigDecimal::class.isSuperclassOf(propertyType)) return BigDecimal(bigInteger)
+            if (BigInteger::class.isSuperclassOf(propertyType)) return bigInteger
+        }
+
+        if (resultSetValue is Number) {
+            val number = resultSetValue
+            if (Byte::class.isSuperclassOf(propertyType)) return number.toByte()
+            if (Short::class.isSuperclassOf(propertyType)) return number.toShort()
+            if (Int::class.isSuperclassOf(propertyType)) return number.toInt()
+            if (Long::class.isSuperclassOf(propertyType)) return number.toLong()
+            if (Float::class.isSuperclassOf(propertyType)) return number.toFloat()
+            if (Double::class.isSuperclassOf(propertyType)) return number.toDouble()
+            if (BigDecimal::class.isSuperclassOf(propertyType)) return BigDecimal(number.toDouble())
+            if (BigInteger::class.isSuperclassOf(propertyType)) return BigDecimal(number.toDouble()).toBigInteger()
+        } else if (resultSetValue is Timestamp) {
+            val date = resultSetValue
+            if (Date::class.isSuperclassOf(propertyType)) return date
+            if (Instant::class.isSuperclassOf(propertyType)) return date.toInstant()
+            if (LocalDate::class.isSuperclassOf(propertyType)) return date.toInstant().atZone(timeZone).toLocalDate()
+            if (LocalDateTime::class.isSuperclassOf(propertyType)) return date.toLocalDateTime()
+        } else if (resultSetValue is java.sql.Date) {
+            val date = resultSetValue
+            if (Date::class.isSuperclassOf(propertyType)) return date
+            if (Instant::class.isSuperclassOf(propertyType)) return date.toInstant()
+            if (LocalDate::class.isSuperclassOf(propertyType)) return date.toLocalDate()
+            if (LocalDateTime::class.isSuperclassOf(propertyType)) return LocalDateTime.ofInstant(date.toInstant(), timeZone)
+        } else if (resultSetValue is java.sql.Time) {
+            if (LocalTime::class.isSuperclassOf(propertyType)) return resultSetValue.toLocalTime()
+        } else if (ZoneId::class.isSuperclassOf(propertyType)) {
+            return ZoneId.of(resultSetValue.toString())
+        } else if (TimeZone::class.isSuperclassOf(propertyType)) {
+            return TimeZone.getTimeZone(resultSetValue.toString())
+        } else if (Locale::class.isSuperclassOf(propertyType)) {
+            return Locale.forLanguageTag(resultSetValue.toString())
+        } else if (Enum::class.isSuperclassOf(propertyType)) {
+            return propertyType.functions.first { func -> func.name == "valueOf" }.call(resultSetValue)
+        } else if ("org.postgresql.util.PGobject" == resultSetValue.javaClass.name) {
```

Comment:
@casey-watson make this `.equals()` just in case...JLS says string literals will work for `==` due to being interned but I am not sure if it guarantees that classnames are interned

Reply:
Kotlin `==` is equivalent to Java `.equals()` 

and kotlin `===` is equivalent to Java `==`

Reply:
That is a very sensible language feature...